templates/cli/src/Client.php.twig

```diff
@@ -0,0 +1,324 @@
+<?php
+
+namespace {{ spec.title | caseUcfirst }};
+
+use Exception;
+use Utopia\CLI\Console;
+
+class Client
+{
+    const METHOD_GET = 'GET';
+    const METHOD_POST = 'POST';
+    const METHOD_PUT = 'PUT';
+    const METHOD_PATCH = 'PATCH';
+    const METHOD_DELETE = 'DELETE';
+    const METHOD_HEAD = 'HEAD';
+    const METHOD_OPTIONS = 'OPTIONS';
+    const METHOD_CONNECT = 'CONNECT';
+    const METHOD_TRACE = 'TRACE';
+
+    const USER_PREFERENCES_FILE = __DIR__."/.prefs.json";
+    const PREFERENCE_ENDPOINT = "endpoint";
+
+    /**
+     * Is Self Signed Certificates Allowed?
+     *
+     * @var bool
+     */
+    private $selfSigned = false;
+
+    /**
+     * Global Headers
+     *
+     * @var array
+     */
+    private $headers = [
+        'content-type' => '',
+        'x-sdk-version' => '{{spec.title | caseDash}}:{{ language.name | caseLower }}:{{ sdk.version }}',
+    ];
+
+    /**
+     * SDK constructor.
+     */
+    public function __construct()
+    {
+        /* Load user defaults from a json file if available
+           Else Propmt the user to enter the details
+        */
+        if (!$this->loadPreferences()) {
+            $this->promptUser();
+        }
+
+        $this
+{% for header in spec.global.headers %}
+{% if header.key != 'Mode' %}
+            ->set{{header.key | caseUcfirst}}($this->preferences['{{header.name}}'])
+{% endif %}
+{% endfor %}
+        ;
+    }
+
+    /**
+     * Default User Preferences
+     *
+     * @var array
+     */
+    private $preferences = [
+        self::PREFERENCE_ENDPOINT => '',
+{% for header in spec.global.headers %}
+{% if header.key != 'Mode' %}
+        '{{header.name}}' => '',
+{% endif %}
+{% endfor %}
+    ];
+
+    public function getPreference(string $key): string
+    {
+        return $this->preferences[$key];
```

Comment:
We should probably make sure key actually exist.

Reply:
Maybe provide a default param & value

Reply:
Done